app/services/planning_application_search.rb

```diff
@@ -8,12 +8,40 @@ class PlanningApplicationSearch
   validates :query, presence: true
 
   def results
-    if valid?
-      planning_applications.where(
-        "LOWER(reference) LIKE ?", "%#{query.downcase}%"
-      )
-    else
-      planning_applications
-    end
+    valid? ? records_matching_query : planning_applications
+  end
+
+  private
+
+  def records_matching_query
+    records_matching_reference.presence || records_matching_description
+  end
+
+  def records_matching_reference
+    planning_applications.where(
+      "LOWER(reference) LIKE ?",
+      "%#{query.downcase}%"
+    )
+  end
+
+  def records_matching_description
+    planning_applications
+      .select("*, ts_rank(#{tsvector}, #{tsquery}) AS rank")
+      .where("#{tsvector} @@ #{tsquery}")
+      .order(rank: :desc)
+  end
+
+  def tsvector
+    "to_tsvector('english', description)"
+  end
+
+  def tsquery
+    ActiveRecord::Base.sanitize_sql_array(
+      ["to_tsquery('english', ?)", query_terms]
+    )
+  end
+
+  def query_terms
+    query.split.join(" | ")
   end
 end
```

Comment:
How about moving `attr_accessor :query, :planning_applications` bellow private?

Also I would be tempted to try and add an initialize method to fetch query param.

How about renaming `query`, to `q`?

Reply:
Ah, so it's set up the way it is because it's (thoughbot's favourite thing 😆) a form object! See https://thoughtbot.com/blog/activemodel-form-objects. 

By having `include ActiveModel::Model` and having `query` and `planning_applications` as public attributes we can treat `PlanningApplicationSearch` in a lot of ways like an active model record - eg [passing it to `form_with`](https://github.com/unboxed/bops/blob/main/app/views/search/_form.html.erb#L2), [intializing it with attributes](https://github.com/unboxed/bops/blob/main/app/controllers/planning_applications_controller.rb#L30), and also running [validations](https://github.com/unboxed/bops/blob/main/app/services/planning_application_search.rb#L8).

Reply:
I marginally prefer `query` to `q` just because stylistically I prefer using the whole word, but happy either way ☺️

Reply:
I prefer query too